crates/extensions/c8y_log_manager/src/actor.rs

```diff
@@ -457,3 +401,351 @@ impl TryIntoOperationStatusMessage for LogfileRequest {
+    use std::path::PathBuf;
+
+    use c8y_api::smartrest::smartrest_deserializer::SmartRestLogRequest;
+    use c8y_http_proxy::messages::C8YRestRequest;
+    use c8y_http_proxy::messages::C8YRestResult;
+    use filetime::set_file_mtime;
+    use filetime::FileTime;
+    use tedge_actors::Actor;
+    use tedge_actors::Builder;
+    use tedge_actors::ReceiveMessages;
+    use tedge_actors::SimpleMessageBox;
+    use tedge_actors::SimpleMessageBoxBuilder;
+    use tedge_mqtt_ext::MqttMessage;
+    use tedge_test_utils::fs::TempTedgeDir;
+    use time::macros::datetime;
+
+    use super::LogManagerActor;
+    use crate::config::FileEntry;
+    use crate::config::LogPluginConfig;
+    use crate::LogManagerBuilder;
+    use crate::LogManagerConfig;
+    use crate::Topic;
+
+    /// Preparing a temp directory containing four files, with
+    /// two types { type_one, type_two }:
+    ///
+    ///     file_a, type_one
+    ///     file_b, type_one
+    ///     file_c, type_two
+    ///     file_d, type_one
+    ///
+    /// each file has the following modified "file update" timestamp:
+    ///     file_a has timestamp: 1970/01/01 00:00:02
+    ///     file_b has timestamp: 1970/01/01 00:00:03
+    ///     file_c has timestamp: 1970/01/01 00:00:11
+    ///     file_d has timestamp: (current, not modified)
+    fn prepare() -> Result<(TempTedgeDir, LogPluginConfig), anyhow::Error> {
+        let tempdir = TempTedgeDir::new();
+        let tempdir_path = tempdir
+            .path()
+            .to_str()
+            .ok_or_else(|| anyhow::anyhow!("temp dir not created"))?;
+
+        std::fs::File::create(format!("{tempdir_path}/file_a"))?;
+        std::fs::File::create(format!("{tempdir_path}/file_b"))?;
+        std::fs::File::create(format!("{tempdir_path}/file_c"))?;
+        std::fs::File::create(format!("{tempdir_path}/file_d"))?;
+
+        let new_mtime = FileTime::from_unix_time(2, 0);
+        set_file_mtime(format!("{tempdir_path}/file_a"), new_mtime).unwrap();
+
+        let new_mtime = FileTime::from_unix_time(3, 0);
+        set_file_mtime(format!("{tempdir_path}/file_b"), new_mtime).unwrap();
+
+        let new_mtime = FileTime::from_unix_time(11, 0);
+        set_file_mtime(format!("{tempdir_path}/file_c"), new_mtime).unwrap();
+
+        let files = vec![
+            FileEntry {
+                path: format!("{tempdir_path}/file_a"),
+                config_type: "type_one".to_string(),
+            },
+            FileEntry {
+                path: format!("{tempdir_path}/file_b"),
+                config_type: "type_one".to_string(),
+            },
+            FileEntry {
+                path: format!("{tempdir_path}/file_c"),
+                config_type: "type_two".to_string(),
+            },
+            FileEntry {
+                path: format!("{tempdir_path}/file_d"),
+                config_type: "type_one".to_string(),
+            },
+        ];
+        let logs_config = LogPluginConfig { files };
```

Comment:
There isn't really any need to hand-craft this `LogPluginConfig` object here like this as it could have been created easily by reading the `c8y-log-plugin.toml` file created below with just `LogPluginConfig::new(tempdir_path.to_path_buf().join("c8y-log-plugin.toml))`. Would be better that way to avoid code duplication and to avoid any divergence.

Reply:
Good point. Fixed.